library/manageiq.py

```diff
@@ -97,8 +116,13 @@ def update_required(self, provider_id, hostname, port):
             result = self.client.get(self.providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
         except Exception as e:
             self.module.fail_json(msg="Failed to get provider data. Error: %s" % e)
-        endpoint = result['endpoints'][0]
-        return False if (endpoint['hostname'] == hostname and endpoint['port'] == int(port)) else True
+
+        def host_port(endpoint):
```

Comment:
suggestion: use lambda function instead of a real inline function:
`host_port = lambda endpoint: (endpoint['hostname'], int(endpoint['port']))`